### PR TITLE
Stagger pusher

### PR DESCRIPTION
**Pull request by Siegrift**
Closes https://github.com/api3dao/signed-api/issues/50 (if this is the implementation we want to go with).

I pushed a docker image on my personal account with this change so it can be tested on AWS https://hub.docker.com/layers/siegrift/api3-pusher-testonly-clone/wip-stagger/images/sha256-45ec5db658f5f1e268ecbb9e1a693d6823b415a49272b49064d9647d5408586c?context=explore

### CODE REVIEW

**Review by metobom — APPROVED** [review]
I've tested this with the Docker [image](https://hub.docker.com/layers/siegrift/api3-pusher-testonly-clone/wip-stagger2/images/sha256-2a51d24e83acc6deaf509bb07880d23eb8d9154a05b97621ca197c73fa32aa73?context=explore) @Siegrift published on his personal accounts.

There are no more timeouts on low cloud resources, it works smoothly.

**Comment by Siegrift** [comment]
> There are no more timeouts on low cloud resources, it works smoothly.

Thanks. Do you have access to [these graphs ](https://github.com/api3dao/signed-api/issues/103#issuecomment-1786744431) ? I'd be interested if the memory and CPU consumption changed somehow.

**Comment by metobom** [comment]
@Siegrift 
newplot

**Comment by Siegrift** [comment]
Thanks, I see that the memory usage dropped significantly (by ~50%) which I don't have explanation for 🤔

File: packages/pusher/src/fetch-beacon-data.ts [hunk 1]
```diff
@@ -42,7 +43,11 @@ const fetchBeaconDataInLoop = async (signedApiUpdate: SignedApiUpdate) => {
       }
     });
     const duration = Date.now() - startTimestamp;
+    // Take at most 10% of the fetch interval as extra time to avoid all API requests be done at the same time. This
+    // delay is taken for each interval, so if the system runs for a sufficiently long time, the requests should happen
+    // at random intervals.
+    const extraTime = Math.random() * signedApiUpdate.fetchInterval * 1000 * 0.1;
```
Comment (bbenligiray) [inline packages/pusher/src/fetch-beacon-data.ts:49]: I was thinking `(Math.random() - 0.5) * signedApiUpdate.fetchInterval * 1000 * 0.1` but this is indeed far nicer

**Pull request merged**